pkg/specs/pods.go

```diff
@@ -381,7 +389,7 @@ func PodWithExistingStorage(cluster apiv1.Cluster, nodeSerial int) *corev1.Pod {
 
 	envConfig := CreatePodEnvConfig(cluster, podName)
 
-	podSpec := CreateClusterPodSpec(podName, cluster, envConfig, gracePeriod)
+	podSpec := CreateClusterPodSpec(podName, cluster, envConfig, gracePeriod, true)
```

Comment:
This is the key entry point. I think a boolean here obscures things.
Might be worth creating a public type here.

Reply:
E.g.
```
type TLScommunication bool

const (
	TLSDisabled TLScommunication = false
	TLSEnabled  TLScommunication = true
)

...
func PodWithExistingStorage(cluster apiv1.Cluster, nodeSerial int) *corev1.Pod {

	podSpec := CreateClusterPodSpec(podName, cluster, envConfig, gracePeriod, TLSEnabled)
```

Reply:
This function is called only in two places. Here, it must always be true, as you cannot opt out of TLS for new pods. It is false only if we compare the generated spec with the spec of an old pod where an old version of the operator is not TLS capable.

Would it be clearer if we create two function

```
func CreateClusterPodSpec(podName, cluster, envConfig, gracePeriod) ...
func CreateClusterPodSpecTLS(podName, cluster, envConfig, gracePeriod, tlsEnabled) ...
```

Reply:
Now that I understand better, I don't think two functions are necessary.
I do think boolean args, sent as "true" always leave you confused, esp with multiple args. What is true?
For the invocation in `PodWithExistingStorage` we could add a var
```
    podSpec := CreateClusterPodSpec(podName, cluster, envConfig, gracePeriod, tlsEnabled)
```
Or, add a comment.